arcaflow_plugin_pcp/pcp_plugin.py

```diff
@@ -218,19 +225,16 @@ def start_pcp(
             # If pcp2json or pcp2csv completes without an exception, we return success.
             return "success", PerfOutput(pcp_metrics_list)
 
-        else:
-            # Return the appropriate error condition after max_retries
-            if "error" in pcp2json_status:
-                return pcp2json_return
-            elif "error" in pcp2csv_status:
-                return pcp2csv_return
-
-            # Since the above for loop should always return either success or error,
-            # and should never come to its natural end, if we get here, something
-            # unexpected went wrong.
-            return "error", Error(
-                "Unknown failure attempting to process pmlogger output"
-            )
+        # Return the appropriate error condition after max_retries
+        if "error" in pcp2json_status:
+            return (pcp2json_status, pcp2json_return)
+        if "error" in pcp2csv_status:
+            return (pcp2csv_status, pcp2csv_return)
+
+        # Since the above for loop should always return either success or error,
+        # and should never come to its natural end, if we get here, something
+        # unexpected went wrong.
```

Comment:
A more pythonic way to trap "natural end" of a loop is with an `else` clause, if you want to push that assumption into code. Or if you really want the fallback of "anytime we get here" (e.g., if someone added a `break`), I'd rephrase this to something more like "exited the loop without returning".

Reply:
I don't really think using an `else` on the `for` loop would help anything, here.  But, I wouldn't mind a rephrase of this comment...eg., since the _loop_ will never result in an _error_ return (we _retry_ on errors...).

But, I would be more inclined to just delete this comment -- I think the code is pretty clear:  we return success from the loop or we retry; when we run out of retries, we return one of two errors; if none of that happens, we return an error pointing out that we had an "Unknown failure" (I might have said "unexpected", but...).

The only real shortcoming here is that, if by some crazy circumstance, we reach the catch-all error return, we don't give any context other than "it didn't work".  (We might be able to find the `pcp2csv_return` from the output at line 202, but we won't find the `pcp2json_return`, as near as I can tell.)